src/main/java/bio/terra/workspace/app/configuration/spring/TraceInterceptorConfig.java

```diff
@@ -0,0 +1,113 @@
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@Configuration
+@ConditionalOnProperty("workspace.tracing.enabled")
+public class TraceInterceptorConfig implements WebMvcConfigurer {
+
+  public static final String MDC_REQUEST_ID_HEADER = "X-Request-ID";
+  public static final String MDC_REQUEST_ID_KEY = "requestId";
+  public static final String MDC_CORRELATION_ID_HEADER = "X-Correlation-ID";
+
+  private final Hashids hashids = new Hashids("requestIdSalt", 8);
+
+  @Autowired
+  public TraceInterceptorConfig(
+      TracingConfiguration tracingConfiguration, Environment environment) {
+    try {
+      StackdriverTraceExporter.createAndRegister(
+          StackdriverTraceConfiguration.builder()
+              .setProjectId(tracingConfiguration.getProjectId())
+              .setCredentials(
+                  ServiceAccountCredentials.fromStream(
+                      new FileInputStream(tracingConfiguration.getSaPath())))
+              .setFixedAttributes(
+                  Map.of(
+                      "/component",
+                      AttributeValue.stringAttributeValue(
+                          environment.getRequiredProperty("spring.application.name"))))
+              .build());
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    TraceConfig globalTraceConfig = Tracing.getTraceConfig();
+    globalTraceConfig.updateActiveTraceParams(
+        globalTraceConfig.getActiveTraceParams().toBuilder()
+            .setSampler(Samplers.probabilitySampler(tracingConfiguration.getProbability()))
+            .build());
+  }
+
+  @Override
+  public void addInterceptors(InterceptorRegistry registry) {
+    registry.addInterceptor(
+        new HandlerInterceptor() {
+          @Override
+          public boolean preHandle(
+              HttpServletRequest httpRequest, HttpServletResponse httpResponse, Object handler)
+              throws Exception {
+
+            // get an mdc id from the request (if not found, create one), and pass it along in the
+            // response
+            String requestId = getMDCRequestId(httpRequest);
+            MDC.put(MDC_REQUEST_ID_KEY, requestId);
+            httpResponse.addHeader(MDC_REQUEST_ID_HEADER, requestId);
+
+            // add tags to Stackdriver traces
+            Tracing.getTracer()
+                .getCurrentSpan()
+                .putAttributes(
+                    Map.of(
+                        MDC_REQUEST_ID_KEY,
+                        AttributeValue.stringAttributeValue(requestId),
+                        "route",
+                        AttributeValue.stringAttributeValue(
+                            ((HandlerMethod) handler)
+                                .getMethodAnnotation(RequestMapping.class)
+                                .path()[0])));
```

Comment:
is there any chance that extraction of the route here will fail? Does this need try/catch or other failsafes?